caddyhttp/fastcgi/fcgiclient.go

```diff
@@ -566,18 +571,20 @@ func (c *FCGIClient) PostFile(p map[string]string, data url.Values, file map[str
 // SetReadTimeout sets the read timeout for future calls that read from the
 // fcgi responder. A zero value for t means no timeout will be set.
 func (c *FCGIClient) SetReadTimeout(t time.Duration) error {
-	c.readTimeout = t
+	if conn, ok := c.rwc.(net.Conn); ok && t > 0 {
```

Comment:
This change really simplifies assigning connect/read/write timeouts! Thank you.

One small observation: The original read/write timeout implementation checked for `timeout != 0` so that deadlines in the past could be used to test handling of the timeout. I borrowed this technique from the [net timeout tests](https://golang.org/src/net/timeout_test.go). I see that this change removes the timeout tests, so it's only something to think about if you decide to reintroduce them. FWIW, I changed `t > 0` to `t != 0` and added `TestReadTimeout` and `TestSendTimeout` tests to my local branch-the tests ultimately passed, but not without encountering [this issue](https://github.com/mholt/caddy/pull/1739#pullrequestreview-47907952).